### DISCUSSION TITLE
Welcome to Advert-Framework Discussions!

### DISCUSSION THREAD

**Discussion by davidhozic** [davidhozic/discord-advertisement-framework#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by TriLogic** [comment]
Does this software shill across multiple servers and how does it find them?

**Reply by davidhozic** [reply]
It does allow to shill across multiple servers, however the framework itself doesn't automatically find them, 
they need to be either added manually 
or with a user callback function:
```py
import framework as fw


servers=[]

async def main():
    user = fw.get_client()
    for guild in user.guilds:
        servers.append(
            fw.GUILD(
                guild_id=guild.id,
                messages_to_send=[
                    fw.TextMESSAGE(None, 10, "Test", 
                                    [
                                        channel.id for channel in guild.channels if "shill" in channel.name
                                    ], 
                                    "send", True)
                ],
                generate_log=True
            )
        )
    pass



fw.run(
    token="token_here",
    server_list=servers,
    is_user=False,
    user_callback=main
)
```
